src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentConstructedWithNoSecureResolverTests.cs

```diff
@@ -0,0 +1,874 @@
+        }
+
+        [Fact]
+        public void XmlDocumentAsFieldSetResolverToNullInSomeMethodShouldGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        public XmlDocument doc = new XmlDocument();
+
+        public void Method1()
+        {
+            this.doc.XmlResolver = null;
+        }
+
+        public void Method2(XmlReader reader)
+        {
+            this.doc.Load(reader);
+        }
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(8, 28, "TestClass")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Public doc As XmlDocument = New XmlDocument()
+
+        Public Sub Method1()
+            Me.doc.XmlResolver = Nothing
+        End Sub
+
+        Public Sub Method2(reader As XmlReader)
+            Me.doc.Load(reader)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(6, 16, "TestClass")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentCreatedAsTempNotSetResolverShouldGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+
+        public void Method1()
+        {
+            Method2(new XmlDocument());
+        }
+
+        public void Method2(XmlDocument doc){}
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(11, 21, "Method1")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+
+        Public Sub Method1()
+            Method2(New XmlDocument())
+        End Sub
+
+        Public Sub Method2(doc As XmlDocument)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(8, 21, "Method1")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentDerivedTypeNotSetResolverShouldNotGenerateDiagnostics()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass1 : XmlDocument
+    {
+        public TestClass1()
+        {
+            XmlResolver = null;
+        }
+    }     
+
+    class TestClass2
+    {
+        void TestMethod()
+        {
+            var c = new TestClass1();
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass1
+        Inherits XmlDocument
+        Public Sub New()
+            XmlResolver = Nothing
+        End Sub
+    End Class
+
+    Class TestClass2
+        Private Sub TestMethod()
+            Dim c = New TestClass1()
+        End Sub
+    End Class
+End Namespace
+"
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentDerivedTypeWithNoSecureResolverShouldNotGenerateDiagnostic()
+        {
```

Comment:
This is desired. This actually verifies 3075 shouldn't check anything done in 3077. 3077 will report on code `class DerivedType : XmlDocument {} `